src/variables/components/TypeAheadVariableDropdown.test.tsx

```diff
@@ -0,0 +1,137 @@
+// Libraries
+import React from 'react'
+import {fireEvent, screen} from '@testing-library/react'
+
+// Components
+import TypeAheadVariableDropdown from 'src/variables/components/TypeAheadVariableDropdown'
+
+// Utils
+import {renderWithRedux} from 'src/mockState'
+import {AppState, RemoteDataState} from 'src/types'
+
+const values = {
+  def: 'defbuck',
+  def2: 'defbuck2',
+  foo: 'foobuck',
+  goo: 'goobuck',
+  new: 'newBuck',
+  always: 'always',
+  really: 'reallyYes',
+  REALLy2: 'anotherReally',
+}
+const fvalues = {
+  def: 'defbuck',
+  def2: 'defbuck2',
+  foo: 'foobuck',
+}
+
+const evalues = {
+  def: 'defbuck',
+  def2: 'defbuck2',
+  new: 'newBuck',
+  really: 'reallyYes',
+  REALLy2: 'anotherReally',
+}
+
+const alvalues = {
+  always: 'always',
+  really: 'reallyYes',
+  REALLy2: 'anotherReally',
+}
+
+const setInitialState = (state: AppState): AppState => {
+  return {
+    ...state,
+    currentDashboard: {
+      id: '03c8070355fbd000',
+    },
+    resources: {
+      ...state.resources,
+      variables: {
+        allIDs: ['03cbdc8a53a63000'],
+        status: RemoteDataState.Done,
+        byID: {
+          '03cbdc8a53a63000': {
+            id: '03cbdc8a53a63000',
+            orgID: '03c02466515c1000',
+            name: 'map_buckets',
+            description: '',
+            selected: null,
+            arguments: {
+              type: 'map',
+              values,
+            },
+            labels: [],
+            status: RemoteDataState.Done,
+          },
+        },
+        values: {
+          '03c8070355fbd000': {
+            status: RemoteDataState.Done,
+            values: {
+              '03cbdc8a53a63000': {
+                values,
+                selected: ['defbuck'],
+              },
+            },
+            order: ['03cbdc8a53a63000'],
+          },
+        },
+      },
+    },
+  }
+}
+
+describe('Dashboards.Components.VariablesControlBar.TypeAheadVariableDropdown', () => {
+  describe('if map type', () => {
```

Comment:
nit:
```suggestion
  describe('the map type', () => {
```

Reply:
i'll change it but it isn't my code (just formatted it with pretty quick)

Reply:
not my code (was just formatted it) but will change it to 'variable map type'